File: scala/src/main/scala/com/spoonacular/client/model/RecipesParseIngredientsNutrition.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.RecipesParseIngredientsNutritionCaloricBreakdown
import com.spoonacular.client.model.RecipesParseIngredientsNutritionNutrients
import com.spoonacular.client.model.RecipesParseIngredientsNutritionProperties
import com.spoonacular.client.model.RecipesParseIngredientsNutritionWeightPerServing
import scala.collection.immutable.Seq

/**
 * 
 * @param nutrients 
 * @param properties 
 * @param flavonoids 
 * @param caloricBreakdown 
 * @param weightPerServing 
 */
case class RecipesParseIngredientsNutrition(nutrients: Seq[RecipesParseIngredientsNutritionNutrients],
                properties: Seq[RecipesParseIngredientsNutritionProperties],
                flavonoids: Seq[RecipesParseIngredientsNutritionProperties],
                caloricBreakdown: RecipesParseIngredientsNutritionCaloricBreakdown,
                weightPerServing: RecipesParseIngredientsNutritionWeightPerServing
                )

object RecipesParseIngredientsNutrition {
    /**
     * Creates the codec for converting RecipesParseIngredientsNutrition from and to JSON.
     */
    implicit val decoder: Decoder[RecipesParseIngredientsNutrition] = deriveDecoder
    implicit val encoder: ObjectEncoder[RecipesParseIngredientsNutrition] = deriveEncoder
}

File: dart/com/spoonacular/client/model/recipes_parse_ingredients_nutrition.dart
part of openapi.api;

class RecipesParseIngredientsNutrition {
  
  List<RecipesParseIngredientsNutritionNutrients> nutrients = [];
  
  List<RecipesParseIngredientsNutritionProperties> properties = [];
  
  List<RecipesParseIngredientsNutritionProperties> flavonoids = [];
  
  RecipesParseIngredientsNutritionCaloricBreakdown caloricBreakdown = null;
  
  RecipesParseIngredientsNutritionWeightPerServing weightPerServing = null;
  RecipesParseIngredientsNutrition();

  @override
  String toString() {
    return 'RecipesParseIngredientsNutrition[nutrients=$nutrients, properties=$properties, flavonoids=$flavonoids, caloricBreakdown=$caloricBreakdown, weightPerServing=$weightPerServing, ]';
  }

  RecipesParseIngredientsNutrition.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['nutrients'] == null) {
      nutrients = null;
    } else {
      nutrients = RecipesParseIngredientsNutritionNutrients.listFromJson(json['nutrients']);
    }
    if (json['properties'] == null) {
      properties = null;
    } else {
      properties = RecipesParseIngredientsNutritionProperties.listFromJson(json['properties']);
    }
    if (json['flavonoids'] == null) {
      flavonoids = null;
    } else {
      flavonoids = RecipesParseIngredientsNutritionProperties.listFromJson(json['flavonoids']);
    }
    if (json['caloricBreakdown'] == null) {
      caloricBreakdown = null;
    } else {
      caloricBreakdown = new RecipesParseIngredientsNutritionCaloricBreakdown.fromJson(json['caloricBreakdown']);
    }
    if (json['weightPerServing'] == null) {
      weightPerServing = null;
    } else {
      weightPerServing = new RecipesParseIngredientsNutritionWeightPerServing.fromJson(json['weightPerServing']);
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (nutrients != null)
      json['nutrients'] = nutrients;
    if (properties != null)
      json['properties'] = properties;
    if (flavonoids != null)
      json['flavonoids'] = flavonoids;
    if (caloricBreakdown != null)
      json['caloricBreakdown'] = caloricBreakdown;
    if (weightPerServing != null)
      json['weightPerServing'] = weightPerServing;
    return json;
  }

  static List<RecipesParseIngredientsNutrition> listFromJson(List<dynamic> json) {
    return json == null ? new List<RecipesParseIngredientsNutrition>() : json.map((value) => new RecipesParseIngredientsNutrition.fromJson(value)).toList();
  }

  static Map<String, RecipesParseIngredientsNutrition> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, RecipesParseIngredientsNutrition>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new RecipesParseIngredientsNutrition.fromJson(value));
    }
    return map;
  }
}

